### PR TITLE
Unittest/testcases

### PR DESCRIPTION
**Pull request by Stark2248**
Added few warehouse service and added test cases for product service

### CODE REVIEW

File: src/main/java/com/silverlining/productmanagement/repository/WarehouseRepository.java [hunk 1]
```diff
@@ -14,6 +15,9 @@
 public interface WarehouseRepository extends JpaRepository<Warehouse,Integer> {
 
 
+    @Query("SELECT w FROM Warehouse w WHERE w.product.serialId = :serialId")
+    List<Warehouse> findBySerialId(@Param("serialId") String serialId);
```
Comment (ishandas387) [inline src/main/java/com/silverlining/productmanagement/repository/WarehouseRepository.java:19]: no need for @Query, JPA will automatically convert. "findBySerialId" to internal query

File: src/main/java/com/silverlining/productmanagement/service/WarehouseServiceImpl.java [hunk 2]
```diff
@@ -30,6 +31,39 @@ public WarehouseServiceImpl(WarehouseRepository warehouseRepository, ProductRepo
...
+    @Override
+    public int getQuantityBySerialIdAndLocation(String serialId, String location) {
+        WarehouseLocation loc = WarehouseLocation.findByName(location);
+        if (loc == null) {
+            return -1;
+        }
+        Optional<Products> productOptional = productRepository.findById(serialId);
+
+        return productOptional.map(product -> {
+            WarehouseDto dto = getWarehouseDto(product, loc);
```
Comment (ishandas387) [inline src/main/java/com/silverlining/productmanagement/service/WarehouseServiceImpl.java:42]: extract to a private method, for clarity

File: src/main/java/com/silverlining/productmanagement/service/WarehouseServiceImpl.java [hunk 3]
```diff
@@ -30,6 +31,39 @@ public WarehouseServiceImpl(WarehouseRepository warehouseRepository, ProductRepo
...
+    @Override
+    public List<WarehouseDto> getProductStock(String serialId) {
+        Optional<Products> optionalProduct = productRepository.findById(serialId);
+        return optionalProduct.map(product ->getWarehouseDtoList(serialId, product)).orElse(Collections.emptyList());
+    }
+
+    private List<WarehouseDto> getWarehouseDtoList(String serialId, Products product) {
+        List<Warehouse> warehouseList = warehouseRepository.findBySerialId(serialId);
+        List<WarehouseDto> dtoList = new ArrayList<>();
+        for(Warehouse warehouse : warehouseList){
```
Comment (ishandas387) [inline src/main/java/com/silverlining/productmanagement/service/WarehouseServiceImpl.java]: Use streams
`warhouseList.stream().map(warehouseItem -> getDTO(warhouseItem)).collect(Collectors.toList())`

**Comment by ishandas387** [comment]
in future, always create branches of feature as feature/abc or not like untitest/

File: src/main/java/com/silverlining/productmanagement/repository/WarehouseRepository.java (hunk 1 above)
Comment (Stark2248) [reply src/main/java/com/silverlining/productmanagement/repository/WarehouseRepository.java:19]: that i tried , got some complicated errors so i converted it to query ..

**Comment by Stark2248** [comment]
Cannot work on both branch at the same time.. or can i?

**Review by ishandas387 — APPROVED** [review]

**Pull request merged**